code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -207,103 +290,111 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
+		. += "Summoner Health: [round(resulthealth, 0.5)]%"
+	if(!COOLDOWN_FINISHED(src, manifest_cooldown))
+		. += "Manifest/Recall Cooldown Remaining: [DisplayTimeText(COOLDOWN_TIMELEFT(src, manifest_cooldown))]"
 
 /mob/living/simple_animal/hostile/guardian/Move() //Returns to summoner if they move out of range
 	. = ..()
 	snapback()
 
 /mob/living/simple_animal/hostile/guardian/proc/snapback()
-	if(summoner)
-		if(get_dist(get_turf(summoner),get_turf(src)) <= range)
-			return
-		else
-			to_chat(src, span_holoparasite("You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]!"))
-			visible_message(span_danger("\The [src] jumps back to its user."))
-			if(istype(summoner.loc, /obj/effect))
-				Recall(TRUE)
-			else
-				new /obj/effect/temp_visual/guardian/phase/out(loc)
-				forceMove(summoner.loc)
-				new /obj/effect/temp_visual/guardian/phase(loc)
+	SIGNAL_HANDLER
+
+	if(!summoner)
+		return
+	if(get_dist(get_turf(summoner), get_turf(src)) <= range)
+		return
+	to_chat(src, span_holoparasite("You moved out of range, and were pulled back! You can only move [range] meters from [summoner.real_name]!"))
+	visible_message(span_danger("\The [src] jumps back to its user."))
+	if(istype(summoner.loc, /obj/effect))
+		Recall(forced = TRUE)
+	else
+		new /obj/effect/temp_visual/guardian/phase/out(loc)
+		forceMove(summoner.loc)
+		new /obj/effect/temp_visual/guardian/phase(loc)
 
 /mob/living/simple_animal/hostile/guardian/canSuicide()
 	return FALSE
 
 /mob/living/simple_animal/hostile/guardian/proc/is_deployed()
 	return loc != summoner
 
-/mob/living/simple_animal/hostile/guardian/AttackingTarget()
+/mob/living/simple_animal/hostile/guardian/AttackingTarget(atom/attacked_target)
 	if(!is_deployed())
-		to_chat(src, "[span_danger("<B>You must be manifested to attack!")]</B>")
+		to_chat(src, span_bolddanger("You must be manifested to attack!"))
 		return FALSE
 	else
 		return ..()
 
-/mob/living/simple_animal/hostile/guardian/death()
-	drop_all_held_items()
-	..()
-	if(summoner)
-		to_chat(summoner, "[span_danger("<B>Your [name] died somehow!")]</B>")
+/mob/living/simple_animal/hostile/guardian/death(gibbed)
+	. = ..()
+	if(!QDELETED(summoner))
+		to_chat(summoner, span_bolddanger("Your [name] died somehow!"))
 		summoner.dust()
 
 /mob/living/simple_animal/hostile/guardian/update_health_hud()
-	if(summoner && hud_used?.healths)
-		var/resulthealth
-		if(iscarbon(summoner))
-			resulthealth = round((abs(HEALTH_THRESHOLD_DEAD - summoner.health) / abs(HEALTH_THRESHOLD_DEAD - summoner.maxHealth)) * 100)
+	if(!summoner)
+		return
+	var/severity = 0
+	var/healthpercent = (iscarbon(summoner) ? (abs(HEALTH_THRESHOLD_DEAD - summoner.health) / abs(HEALTH_THRESHOLD_DEAD - summoner.maxHealth)) : (summoner.health / summoner.maxHealth)) * 100
```

Comment:
similar. tho if this is the same logic as before why not move it out to a different helper proc?